### PR TITLE
ABB IRB6700 support package

### PR DESCRIPTION
**Pull request by marip8**
Support package for the 235 2.65 variant of the ABB IRB6700

### CODE REVIEW

File: abb_irb6700_support/package.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" ?>
+<package>
```
Comment (gavanderhoorn) [inline abb_irb6700_support/package.xml]: Could we please make this a "format 2" manifest?

Differences will be minor.

Refer to [abb_irb2600_support/package.xml](https://github.com/ros-industrial/abb_experimental/blob/d6900d8b56913a10558db6ca1e0e6a68d7896690/abb_irb2600_support/package.xml) for an example.

File: abb_irb6700_support/package.xml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  <name>abb_irb6700_support</name>
+  <version>0.1.0</version>
+  <description>
+    <p>
+      ROS-Industrial support for the ABB IRB 6700 (and variants).
+    </p>
+    <p>
+      This package contains configuration data, 3D models and launch files
+      for ABB IRB 52 manipulators. This currently includes the 7/1.2 and
+      7/1.45 variants.
```
Comment (gavanderhoorn) [inline abb_irb6700_support/package.xml]: copy-pasta?

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    <joint name="${prefix}joint_1" type="revolute">
+      <origin xyz="0 0 0.288" rpy="0 0 0" />
+      <parent link="${prefix}base_link" />
+      <child link="${prefix}link_1" />
+      <axis xyz="0 0 1" />
+      <limit lower="-${radians(170)}" upper="${radians(170)}" effort="0" velocity="${radians(100)}" />
+    </joint>
+    <joint name="${prefix}joint_2" type="revolute">
+      <origin xyz="0.32 0 0.492" rpy="0 0 0" />
```
Comment (gavanderhoorn) [inline abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: This transform has translations in two dimensions. It's preferred to have each `joint` translate in only a single dimension.

Could you please update this `joint` (and the mesh) to make sure it translates over only a single axis?

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+      <limit lower="-${radians(130)}" upper="${radians(130)}" effort="0" velocity="${radians(120)}" />
+    </joint>
+    <joint name="${prefix}joint_6" type="revolute">
+      <origin xyz="0.165 0 0" rpy="0 0 0" />
+      <parent link="${prefix}link_5" />
+      <child link="${prefix}link_6" />
+      <axis xyz="1 0 0" />
+      <limit lower="-${radians(360)}" upper="${radians(360)}" effort="0" velocity="${radians(190)}" />
+    </joint>
+    <joint name="${prefix}tool_joint" type="fixed">
```
Comment (gavanderhoorn) [inline abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]:
```suggestion
    <joint name="${prefix}joint6-tool0" type="fixed">
```

Comment (gavanderhoorn) [reply abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: These joints typically have zero-translations associated with them (it's an unconfigured `tool0` link).

Is this robot different, or the xacro?

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+      <mimic joint="${prefix}joint_2" multiplier="-0.25"/>
+      <limit lower="-3" upper="3" effort="0" velocity="100" />
+    </joint>
+    <joint name="${prefix}piston_joint" type="prismatic">
+      <origin xyz="0 0 0" rpy="0.20897 0 -${radians(90)}" />
+      <parent link="${prefix}cylinder" />
+      <child link="${prefix}piston" />
+      <axis xyz="0 1 0" />
+      <mimic joint="${prefix}joint_2" multiplier="0.15"/>
+      <limit lower="-3" upper="3" effort="0" velocity="100" />
```
Comment (gavanderhoorn) [inline abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: I know these limits don't really matter (same for the `revolute` joint above), but did you just make these up or actually find some reasonable nrs for them?

Comment (gavanderhoorn) [reply abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: Also `(-)3` would be 3 metres. That seems a lot for a robot this size.

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    </joint>-->
+
+    <!-- ROS base_link to ABB World Coordinates transform -->
+    <link name="${prefix}base" />
+    <joint name="${prefix}base_link_base" type="fixed">
+      <origin xyz="0 0 0.780" rpy="0 0 0"/>
```
Comment (gavanderhoorn) [inline abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: Is this correct? Other xacros in this repository (and over at `ros-industrial/abb`) have an all-zeros transform here typically.

**Review by gavanderhoorn — CHANGES_REQUESTED** [review]
Some comments. Minor changes requested.

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro (hunk 5 above)
Comment (marip8) [reply abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: I just made an estimate for the limits and the mimic multiplier; I'll reduce the limits to more reasonable values

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro (hunk 6 above)
Comment (marip8) [reply abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: I think this was a mistake from working with other vendors' robots; I'll change it to match the other ABB models until I can verify on actual hardware

File: abb_irb6700_support/urdf/irb6700_235_265_macro.xacro (hunk 4 above)
Comment (marip8) [reply abb_irb6700_support/urdf/irb6700_235_265_macro.xacro]: This was just bad xacro design; I updated the link_6 model and changed the tool frame to have zero translation

**Comment by gavanderhoorn** [comment]
@marip8: what is the status of this PR?

**Comment by marip8** [comment]
@gavanderhoorn The changes you requested are addressed in the last 2 commits, so it should be good to merge. Not sure where the merge commit came from; I can try to rebase or squash that commit if necessary

**Comment by gavanderhoorn** [comment]
> @gavanderhoorn The changes you requested are addressed in the last 2 commits, so it should be good to merge. Not sure where the merge commit came from; I can try to rebase or squash that commit if necessary

I'll squash-merge.

Just fixed a minor issue. Let's wait on Travis and we'll merge.

**Review by gavanderhoorn — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
Travis completed successfully (but took its time), so merging.

**Comment by gavanderhoorn** [comment]
Thanks @marip8 for submitting the PR :+1: .

**Pull request merged**